src/main/java/com/eternalcode/parcellockers/parcel/repository/ParcelRepository.java

```diff
@@ -9,16 +11,19 @@ public interface ParcelRepository {
 
     CompletableFuture<Void> save(Parcel parcel);
 
-    // Updates the content and metadata of the parcel, however, it does not replace the UUID
     CompletableFuture<Void> update(Parcel oldParcel, Parcel newParcel);
 
-    CompletableFuture<Optional<Parcel>> findByUuid(UUID uuid);
+    CompletableFuture<Optional<Parcel>> findByUUID(UUID uuid);
 
-    CompletableFuture<Set<Parcel>> findAll();
+    CompletableFuture<Set<Parcel>> findBySender(UUID sender);
 
-    CompletableFuture<Set<Parcel>> findBySender(UUID uuid);
+    CompletableFuture<Set<Parcel>> findByReceiver(UUID receiver);
+
+    CompletableFuture<Set<Parcel>> findAll();
```

Comment:
do wywalenia, po to zrobiłeś findPage aby nie iterować całej bazy danych

Reply:
Póki co wstrzymuję się z usuwaniem tej metody, zamierzam to robić na bieżąco w następnych PR-kach od kodu biznesowego